spec/facility_spec.rb

```diff
@@ -23,4 +23,165 @@
       expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
     end
   end
+
+  describe 'registered_vehicles' do
+    it 'can list registered vehicles' do
+      expect(@facility.registered_vehicles).to eq([])
+      @facility.add_service('Vehicle Registration')
+      
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      bolt = Vehicle.new({vin: '987654321abcdefgh', year: 2019, make: 'Chevrolet', model: 'Bolt', engine: :ev} )
+
+      @facility.register_vehicle(cruz)
+      @facility.register_vehicle(bolt)
+      expect(@facility.registered_vehicles).to eq([cruz, bolt])
+    end
+  end
+
+  describe 'register_vehicle' do
+    it 'can register a vehicle' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+
+      @facility.add_service('Vehicle Registration')
+      @facility.register_vehicle(cruz)
+
+      expect(@facility.registered_vehicles).to eq([cruz])
+      expect(@facility.collected_fees).to eq(100)
+      expect(cruz.plate_type).to eq("regular")
+      expect(cruz.registration_date).to eq(Date.today)
+    end
+  end
+
+  describe 'registration_fee' do
+    it 'can calculate registration fee and update collected fees' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+
+      @facility.registration_fee(cruz)
+
+      expect(@facility.collected_fees).to eq(100)
+    end
+  end
+
+  describe 'update_plate_type' do
+    it 'can update plate type' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+
+      cruz.update_plate_type(cruz)
+
+      expect(cruz.plate_type).to eq('regular')
+    end
+  end
+
+  describe 'update_registered_vehicles' do
+    it 'can update array of vehicles' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+
+      @facility.update_registered_vehicles(cruz)
```

Comment:
This test could be more robust if it tries to register a vehicle and verifies that the vehicle has been added to the registered_vehicles array (rather than just calling the helper method that just shovels the vehicle into the array)